src/status_im/extensions/registry.cljs

```diff
@@ -143,22 +143,18 @@
                                      :ephemeral? ephemeral?}
                         :active? true}
         new-extensions (assoc (:extensions account) url extension)]
-    (if ephemeral?
-      (fx/merge cofx
-                {:dispatch (if modal?
-                             [:navigate-back]
-                             [:navigate-to-clean :my-profile])}
-                (when on-activation (on-activation)))
-      (fx/merge cofx
-                {:utils/show-popup {:title      (i18n/label :t/success)
-                                    :content    (i18n/label :t/extension-installed)
-                                    :on-dismiss #(re-frame/dispatch (if modal?
-                                                                      [:navigate-back]
-                                                                      [:navigate-to-clean :my-profile]))}}
-                (when hooks (accounts.update/account-update {:extensions new-extensions} {}))
-                (when hooks (add-to-registry url extension-data true))
-                (when on-installation (on-installation))
-                (when on-activation (on-activation))))))
+    (do
+      (re-frame/dispatch (if modal?
+                           [:navigate-back]
+                           [:navigate-to-clean :my-profile]))
+      (if ephemeral?
+        (fx/merge cofx
```

Comment:
Can the if be moved inside the merge?

Reply:
not really sure we can get compact version with the condition inside.. please guide me if have a good idea :)

Reply:
Right probably simpler the way you did it.

Reply:
@jeluard i merged the merges 😄